src/Cli/dotnet/commands/dotnet-workload/restore/WorkloadRestoreCommandParser.cs

```diff
@@ -8,21 +8,12 @@ namespace Microsoft.DotNet.Cli
 {
     internal static class WorkloadRestoreCommandParser
     {
-        public static readonly Option ConfigOption = WorkloadInstallCommandParser.ConfigOption;
-
-        public static readonly Option SourceOption = WorkloadInstallCommandParser.SourceOption;
-
-        public static readonly Option VerbosityOption = WorkloadInstallCommandParser.VerbosityOption;
-
         public static Command GetCommand()
         {
             Command command = new Command("restore", LocalizableStrings.CommandDescription);
 
-            command.AddOption(ConfigOption);
-            command.AddOption(SourceOption);
-            command.AddWorkloadCommandNuGetRestoreActionConfigOptions();
-            command.AddOption(VerbosityOption);
-
+            command.AddArgument(RestoreCommandParser.SlnOrProjectArgument);
```

Comment:
I use an msbuild API to read sln to projects

Reply:
Since this PR parses the sln it's not critical but

> I don't know off the top of my head what will happen if you call an arbitrary target in a solution.

Since https://github.com/dotnet/msbuild/pull/1497, MSBuild will synthesize a target that calls the arbitrary target on each project in the solution (but today we don't roll up the results into an observable target output).